### PR TITLE
Ignore CERT_ALREADY_IN_HASH_TABLE error

### PR DESCRIPTION
**Pull request by sbko**
Check ec value for CERT_ALREADY_IN_HASH_TABLE instead of using ERR_get_error.

### CODE REVIEW

File: src/common/http/http.cc [hunk 1]
```diff
@@ -424,11 +424,11 @@ response_t HttpImpl::Post(
...
-        auto err = ERR_get_error();
+        std::string ca_message = ca_ec.message();
+        bool hash_error = ca_message.find("CERT_ALREADY_IN_HASH_TABLE") != std::string::npos;
         // We can ignore this error. Reference:
         // https://github.com/facebook/folly/blob/d3354e2282303402e70d829d19bfecce051a5850/folly/ssl/OpenSSLCertUtils.cpp#L367-L368.
-        if (ERR_GET_LIB(err) != ERR_LIB_X509 ||
-            ERR_GET_REASON(err) != X509_R_CERT_ALREADY_IN_HASH_TABLE) {
+        if (!hash_error) {
           throw boost::system::system_error{ca_ec};
         }
```
Comment (dio) [inline src/common/http/http.cc:433]: Probably something like:

```cpp
        // X509_R_CERT_ALREADY_IN_HASH_TABLE can be ignored.
        // Reference:
        // https://github.com/facebook/folly/blob/d3354e2282303402e70d829d19bfecce051a5850/folly/ssl/OpenSSLCertUtils.cpp#L367-L368.
        if (ca_ec.category() != boost::asio::error::get_ssl_category() ||
            ERR_GET_REASON(ca_ec.value()) !=
                X509_R_CERT_ALREADY_IN_HASH_TABLE) {
          throw boost::system::system_error{ca_ec};
        }
```

I'm checking this on my local.

Comment (dio) [reply src/common/http/http.cc:433]: So the asio's `ctx.add_certificate_authority` has already done the `ERR_get_error`, and wrapped it into an `error_code` "object".

Comment (dio) [reply src/common/http/http.cc:433]: @sbko when you have time, could you help to test: https://github.com/3128px/authservice/releases/download/0.5.3-test2/auth_server_linux_amd64_clang-fips_0.5.3-test2.tar.gz? Thanks!

Comment (sbko) [reply src/common/http/http.cc:433]: works!

**Comment by sbko** [comment]
closed in favor of #230

**Pull request closed without merging**